lib/ansible/plugins/strategy/__init__.py

```diff
@@ -481,6 +481,7 @@ def parent_handler_match(target_handler, handler_name):
                             item = result_item.get(loop_var, None)
                             if item is not None:
                                 task_vars[loop_var] = item
+                                templar.set_available_variables(task_vars)  # invalidates cache
```

Comment:
@bcoca After #22003 I don't see where "{{ item }}" is being replaced by the actual host: `target_host` [here](https://github.com/ansible/ansible/pull/22003/files#diff-b893b8b8319cd4518208dd4fe53f3d7bR507) only receives "{{ item }}".

Output of the playbook from the original report on latest devel is as follows:

```
TASK [Gather facts] ************************************************************************************************************************
ok: [server-3 -> 2001:db8:::1] => (item=server-1)
skipping: [server-3] => (item=server-3) 
ok: [server-3 -> 2001:db8:::2] => (item=server-2)

TASK [debug] *******************************************************************************************************************************
fatal: [server-3]: FAILED! => {"failed": true, "msg": "the field 'args' has an invalid value, which appears to include a variable that is undefined. The error was: 'dict object' has no attribute 'ansible_fqdn'\n\nThe error appears to have been in '/xxxx/playbook.yml': line 16, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n    - debug: msg=\"{{ hostvars[item]['ansible_fqdn'] }}\"\n      ^ here\nWe could be wrong, but this one looks like it might be an issue with\nmissing quotes.  Always quote template expression brackets when they\nstart a value. For instance:\n\n    with_items:\n      - {{ foo }}\n\nShould be written as:\n\n    with_items:\n      - \"{{ foo }}\"\n"}
```

I will work on a new patchset.